src/features/counter/comments.ts

```diff
@@ -0,0 +1,42 @@
+import { PayloadAction, createSlice } from '@reduxjs/toolkit';
+import { Comment } from '../../types/Comment';
+
+export type CommentsState = {
+  comments: Comment[];
+  loading: boolean;
+  error: string;
+};
+
+const initialState: CommentsState = {
+  comments: [],
+  loading: false,
+  error: '',
+};
+
+export const commentsSlice = createSlice({
+  name: 'comments',
+  initialState,
+  reducers: {
+    setComments: (state, action: PayloadAction<Comment[]>) => {
+      state.comments = action.payload;
+    },
+    addComments: (state, action) => {
```

Comment:
use async thunk (`createAsyncThunk`) to handle necessary promise states in extra reducers